docs/upgrading_to_v3.0.md

```diff
@@ -0,0 +1,34 @@
+# Upgrading to v3.0
```

Comment:
The new version will be v4.0.

Reply:
Done. I thought since v3 was so recent that it would be ok to put it under there, but makes sense to make it v4.

Can you explain why some fields despite being bools are still interpreted as strings? Is it due to templating? Seems a bit confusing and error prone. If we're making a v4 jump I can also convert all string bools to actual bools as well.

Reply:
I think that's just legacy, we can ignore for now.